lib/roby/app/rake.rb

```diff
@@ -576,7 +668,68 @@ def use_junit?
                 # Path to the JUnit/Rubocop reports (if enabled)
                 attr_accessor :report_dir
 
-                def run_roby_test(*args, report_name: "report", coverage_name: "roby")
+                def write_captured_output_sync(success, output, omit_tests_success)
+                    Rake.report_sync_mutex.synchronize do
+                        puts output unless omit_tests_success && success
+                    end
+                end
+
+                def write_captured_output(
+                    success, output, synchronize_output, omit_tests_success
+                )
+                    if synchronize_output
+                        write_captured_output_sync(success, output, omit_tests_success)
+                    else
+                        puts output unless omit_tests_success && success
+                    end
+                end
+
+                def spawn_process_capturing_output(bin, *args)
+                    stdout_r, stdout_w = IO.pipe
+                    pid = spawn(Gem.ruby, bin, *args, out: stdout_w, err: stdout_w)
+                    stdout_w.close
+                    [pid, stdout_r]
+                end
+
+                def read_captured_output_from_pipe(pid, read_pipe)
+                    output = []
+                    begin
+                        while (output_fragment = read_pipe.read(512))
+                            output << output_fragment
+                        end
+                        output.join ""
+                    rescue Interrupt
+                        Process.kill "INT", pid
+                        Process.waitpid pid
+                    end
+                end
+
+                def wait_process_with_captured_output(
+                    pid, read_pipe, synchronize_output:, omit_success:
+                )
+                    output = read_captured_output_from_pipe(pid, read_pipe)
+                    _, status = Process.waitpid2(pid)
+                    success = status.success?
+                    write_captured_output(
+                        success, output, synchronize_output, omit_success
+                    )
+                    puts "#{task_name} tests succeeded.\n\n" if success
+                    success
+                end
+
+                def spawn_process(bin, *args)
+                    pid = spawn(Gem.ruby, bin, *args)
+                    begin
+                        _, status = Process.waitpid2(pid)
+                        status.success?
+                    rescue Interrupt
+                        Process.kill "INT", pid
+                        Process.waitpid pid
+                    end
+                end
```

Comment:
why don't put this in a common module?

Reply:
The goal for RobotTestTask is to be a simpler implementation for TestTask. It is mostly a copy of TestTask right now, without some less useful features. Since we plan on eventually replacing TestTask for RobotTestTask, we didnt bother to create a proper structure with shared methods, as when this time comes TestTask will be deprecated.